pilot/pkg/xds/filters/filters.go

```diff
@@ -436,3 +352,132 @@ var (
 		TypedConfig: protoconv.MessageToAny(&resourcedetectors.DynatraceResourceDetectorConfig{}),
 	}
 )
+
+func TCPClusterMx(additionalLabels ...string) *cluster.Filter {
+	cfg := map[string]any{
+		"protocol":         "istio-peer-exchange",
+		"enable_discovery": true,
+	}
+	if len(additionalLabels) != 0 {
+		cfg["additional_labels"] = convertToAnySlice(additionalLabels...)
+	}
+
+	return &cluster.Filter{
+		Name:        MxFilterName,
+		TypedConfig: protoconv.TypedStructWithFields(MetadataExchangeTypeURL, cfg),
+	}
+}
+
+func convertToAnySlice(s ...string) []any {
+	res := make([]any, 0, len(s))
+	for _, v := range s {
+		res = append(res, v)
+	}
+	return res
+}
+
+func TCPMx(additionalLabels ...string) *listener.Filter {
+	cfg := map[string]any{
+		"protocol":         "istio-peer-exchange",
+		"enable_discovery": true,
+	}
+	if len(additionalLabels) != 0 {
+		cfg["additional_labels"] = convertToAnySlice(additionalLabels...)
+	}
+
+	return &listener.Filter{
+		Name: MxFilterName,
+		ConfigType: &listener.Filter_TypedConfig{
+			TypedConfig: protoconv.TypedStructWithFields(MetadataExchangeTypeURL, cfg),
+		},
+	}
+}
+
+func SidecarInboundMx(additionalLabels ...string) *hcm.HttpFilter {
+	pmCfg := map[string]any{
+		"downstream_discovery": []any{
+			map[string]any{
+				"istio_headers": map[string]any{},
+			},
+			map[string]any{
+				"workload_discovery": map[string]any{},
+			},
+		},
+		"downstream_propagation": []any{
+			map[string]any{
+				"istio_headers": map[string]any{},
+			},
+		},
+	}
+	if len(additionalLabels) != 0 {
+		pmCfg["additional_labels"] = convertToAnySlice(additionalLabels...)
+	}
+
+	return &hcm.HttpFilter{
+		Name: MxFilterName,
+		ConfigType: &hcm.HttpFilter_TypedConfig{
+			TypedConfig: protoconv.TypedStructWithFields(PeerMetadataTypeURL, pmCfg),
+		},
+	}
+}
+
+func SidecarOutboundMx(additionalLabels ...string) *hcm.HttpFilter {
+	pmCfg := map[string]any{
+		"upstream_discovery": []any{
+			map[string]any{
+				"istio_headers": map[string]any{},
+			},
+			map[string]any{
+				"workload_discovery": map[string]any{},
+			},
+		},
+		"upstream_propagation": []any{
+			map[string]any{
+				"istio_headers": map[string]any{},
+			},
+		},
+	}
+	if len(additionalLabels) != 0 {
+		pmCfg["additional_labels"] = convertToAnySlice(additionalLabels...)
```

Comment:
why do we have to convert? its a `map[string]any`, we can just put a `[]string` as the any we don't need a `[]any`? It should serialize the same?

Reply:
I recall `TypedStructWithFields` will report warning without this conversion.